src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -51,10 +61,71 @@ public boolean retainAll(Collection<?> c) {
     }
 
     public boolean removeAll(Collection<?> c) {
-        return false;
+        Objects.requireNonNull(c);
+        boolean var2 = false;
+        Iterator var3;
+        if (this.size() > c.size()) {
+            for(var3 = c.iterator(); var3.hasNext(); var2 |= this.remove(var3.next())) {
+            }
+        } else {
+            var3 = this.iterator();
+
+            while(var3.hasNext()) {
+                if (c.contains(var3.next())) {
+                    var3.remove();
```

Comment:
why do you remove from collection? You have to remove from map.
Review method logic.